cmds/ocm/commands/ocmcmds/common/handlers/vershdlr/typehandler.go

```diff
@@ -0,0 +1,234 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package vershdlr
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Masterminds/semver/v3"
+
+	"github.com/open-component-model/ocm/cmds/ocm/pkg/output"
+	"github.com/open-component-model/ocm/cmds/ocm/pkg/utils"
+	"github.com/open-component-model/ocm/pkg/contexts/clictx"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm"
+	"github.com/open-component-model/ocm/pkg/errors"
+	"github.com/open-component-model/ocm/pkg/semverutils"
+)
+
+func Elem(e interface{}) *Object {
+	return e.(*Object)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+type Object struct {
+	Spec       ocm.RefSpec
+	Repository ocm.Repository
+	Component  string
+	Version    string
+}
+
+type Manifest struct {
+	Component string `json:"component"`
+	Version   string `json:"version"`
+	Message   string `json:"error,omitempty"`
+}
+
+func (o *Object) AsManifest() interface{} {
+	tag := ""
+	msg := ""
+	if o.Spec.Version != nil {
+		tag = *o.Spec.Version
+	}
+	if o.Version == "" {
+		msg = "<unknown component version>"
+	}
+	return &Manifest{
+		o.Component,
+		tag,
+		msg,
+	}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+type TypeHandler struct {
+	octx        clictx.OCM
+	session     ocm.Session
+	repobase    ocm.Repository
+	resolver    ocm.ComponentVersionResolver
+	constraints []*semver.Constraints
+	latest      bool
+}
+
+func NewTypeHandler(octx clictx.OCM, session ocm.Session, repobase ocm.Repository, opts ...Option) utils.TypeHandler {
+	h := &TypeHandler{
+		octx:     octx,
+		session:  session,
+		repobase: repobase,
+	}
+	for _, o := range opts {
+		o.ApplyToHandler(h)
+	}
+	return h
+}
+
+func (h *TypeHandler) Close() error {
+	return nil
+}
+
+func (h *TypeHandler) All() ([]output.Object, error) {
+	if h.repobase == nil {
+		return nil, nil
+	}
+	return h.all(h.repobase)
+}
+
+func (h *TypeHandler) all(repo ocm.Repository) ([]output.Object, error) {
+	lister := repo.ComponentLister()
+	if lister == nil {
+		return nil, nil
+	}
+	list, err := lister.GetComponents("", true)
+	if err != nil {
+		return nil, err
+	}
+	var result []output.Object
+	for _, l := range list {
+		part, err := h.get(repo, utils.StringSpec(l))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Warning: %s\n", err)
+		}
+		result = append(result, part...)
+	}
+	return result, nil
+}
+
+func (h *TypeHandler) Get(elemspec utils.ElemSpec) ([]output.Object, error) {
+	return h.get(h.repobase, elemspec)
+}
+
+func (h *TypeHandler) filterVersions(vers []string) []string {
+	if len(h.constraints) == 0 && !h.latest {
+		return vers
+	}
+	versions, _ := semverutils.MatchVersionStrings(vers, h.constraints...)
+	if h.latest && len(versions) > 1 {
+		versions = versions[len(versions)-1:]
```

Comment:
might work only correctly, when NOT sorted by string comparison, but Major.Minor.Patch semver comparison